contrib/gorilla/websocket/websocket.go

```diff
@@ -0,0 +1,143 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+// Package websocket provides tracing functions for tracing the gorilla/websocket
+// package (https://github.com/gorilla/websocket).
+package websocket // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/gorilla/websocket"
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Upgrader is the wrapper type of a traced upgrader.
+type Upgrader struct {
+	websocket.Upgrader
+}
+
+// WrapUpgrader wraps the upgrader traced with the global tracer.
+func WrapUpgrader(u websocket.Upgrader) Upgrader {
+	return Upgrader{Upgrader: u}
+}
+
+// Upgrade traces the actual Upgrade() method using the global tracer as span
+// websocket.connection, which finishes when the returned connection is closed,
+// or in case of an error.
+func (u *Upgrader) Upgrade(w http.ResponseWriter, r *http.Request, responseHeader http.Header) (conn *Conn, err error) {
+	span, ctx := tracer.StartSpanFromContext(r.Context(), "websocket.connection")
+	defer func() {
+		// Finish the span when an error occurred.
+		// Otherwise, finish it in the (*Conn).Close() method.
+		if err != nil {
+			span.Finish(tracer.WithError(err))
+		}
+	}()
+
+	tracee, err := u.Upgrader.Upgrade(w, r.WithContext(ctx), responseHeader)
```

Comment:
```suggestion
	conn, err := u.Upgrader.Upgrade(w, r.WithContext(ctx), responseHeader)
```
Why not `conn`?

Reply:
the new WarpConn api now requires handling two conn variables in the same scope (unless writing some more line to have a separate scope - but good enough for this small example)